viewer/packages/360-images/src/Image360Facade.ts

```diff
@@ -37,8 +37,14 @@ export class Image360Facade<T> {
     this.getCollectionContainingEntity(entity).setSelectedVisibility(visible);
   }
 
-  hideAllHoverIcons(): void {
-    this._image360Collections.forEach(collection => collection.setSelectedVisibility(false));
+  hideAllHoverIcons(): boolean {
+    let changed = false;
+    for (const collection of this._image360Collections) {
```

Comment:
We can use `Array.some()` method instead of `for` here like below, it will also remove temp variable as well
```
hideAllHoverIcons(): boolean {
  return this._image360Collections.some(collection => collection.setSelectedVisibility(false));
}
```

Reply:
You cant'. I have to call setSelectedVisibility() for all anyway

Reply:
I get that you have to set visibility for all objects in collection than, we can use Array.reduce() get the required results as below
```
hideAllHoverIcons(): boolean {
  return this._image360Collections.reduce((changed, collection) => {
    return collection.setSelectedVisibility(false) || changed;
  }, false);
}
```